### PR TITLE
UnitTests: Login/ManageAccount(Cre/Edi/Del) , Changes to functions.py structure and added more functions to reduce code in views.py

### PR DESCRIPTION
**Pull request by YuriKleyman**

### CODE REVIEW

**Comment by YuriKleyman** [comment]
going to finish login tests tomorrow with more test cases

File: CS361_Project/tests/unitTests/test_Login.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        # Assert that session variables are set correctly
+        self.assertEqual(mock_request.session['name'], self.mock_user.name)
+        self.assertEqual(mock_request.session['role'], self.mock_user.role)
+        self.assertTrue(mock_request.session['LoggedIn'])
+    def test_authenticate_user(self):
+         mock_user = Mock()
+         mock_user.username = "test_user"
+         mock_user.password = "test_password"
+         #TODO: not sure if this is even correct, any other way to mock an account that will show in get?
+         Account.objects.get = Mock(return_value=mock_user)
```
Comment (YuriKleyman) [inline CS361_Project/tests/unitTests/test_Login.py:33]: Can someone check this? Im not sure if this is correct

**Review by AaronMcGarry — APPROVED** [review]
lookin good, hopefully the gitignore fixes manageaccount

File: CS361_Project/tests/unitTests/test_Login.py (hunk 1 above)
Comment (YuriKleyman) [reply CS361_Project/tests/unitTests/test_Login.py]: just resolved the issue, turns out Account.objects.get = Mock(return_value=mock_user) works just fine. All the login unit tests are passing and the other unit tests are mostly passing, ill try to finish the rest of the manageAccount tests today.

**Review by ILoveJRock — APPROVED** [review]
Test driven development my beloved

File: templates/Manageaccount.html
Comment (ensonnen) [inline templates/Manageaccount.html:1]: my only concern is that there's script here

**Review by ensonnen — APPROVED** [review]
looks good, only thing is that theres script in manage account template?? thought that wasn't allowed

**Pull request merged**